### ISSUE TITLE
default behavior for `name` in `read_xarray` can lead to unexpected behavior

### ISSUE THREAD

**Issue by neptunes5thmoon** [janelia-cellmap/fibsem-tools#71]
Dask identifies which computations are unique based on the `name` of the dask array. If I don't specify `name` when calling `read_xarray`, for zarr the `name` of the dask array gets set to `element.basename` [here](https://github.com/janelia-cellmap/fibsem-tools/blob/2d0ddae68ce5fd08d43e6158091d5fa6817593a3/src/fibsem_tools/io/zarr.py#L488C32-L488C32). This ends up being the very last bit of the path.

In my use case I'm reading a lot of different arrays that are saved as multiscales and then stacking them together. I then call `.compute()` on this stacked array. However, all the arrays share the same `name` (in my case `"s1"`) and dask collapses that computation and I end up with a stacked array that has the same data in all of the slices.
Everything works as expected if I set `name` explicitly to the full path - maybe this would also be a better default?

**Comment by d-v-b** [comment]
yeah i think this makes sense, I think my logic for setting the name of the dask array to `element.basename` wasn't  actually sound in the first place, so maybe we just avoid setting the name field at all? Alternatively, if we think the name of the array is useful for something, we do something like `<name>-<hash>`, where `<hash>` is uniquely generated each invocation. but maybe this is overthinking it 😅

**Comment by neptunes5thmoon** [comment]
I think letting dask do the hashing is probably good. If setting the name is useful for something then people can still set it themselves

**Comment by d-v-b** [comment]
this should be fixed by #72
